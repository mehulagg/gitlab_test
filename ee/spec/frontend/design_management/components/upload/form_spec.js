import { shallowMount } from '@vue/test-utils';
import UploadForm from 'ee/design_management/components/upload/form.vue';

describe('Design management upload form component', () => {
  let vm;

  function createComponent(isSaving = false, canUploadDesign = true) {
    vm = shallowMount(UploadForm, {
      propsData: {
        isSaving,
        canUploadDesign,
      },
    });
  }

  it('renders upload design button', () => {
    createComponent();

    expect(vm.element).toMatchSnapshot();
  });

  it('renders loading icon', () => {
    createComponent(true);

    expect(vm.element).toMatchSnapshot();
  });

  it('hides button if cant upload', () => {
    createComponent(false, false);

    expect(vm.element).toMatchSnapshot();
  });

  describe('onFileUploadChange', () => {
    it('emits upload event', () => {
      createComponent();

      jest.spyOn(vm.find({ ref: 'fileUpload' }).element, 'files', 'get').mockReturnValue('test');

      vm.vm.onFileUploadChange();

      expect(vm.emitted().upload[0]).toEqual(['test']);
    });
  });

  describe('openFileUpload', () => {
    it('triggers click on input', () => {
      createComponent();

      const clickSpy = jest.spyOn(vm.find({ ref: 'fileUpload' }).element, 'click');

      vm.vm.openFileUpload();

      expect(clickSpy).toHaveBeenCalled();
    });
  });
});
