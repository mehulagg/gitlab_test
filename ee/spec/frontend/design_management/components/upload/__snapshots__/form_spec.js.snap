// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management upload form component hides button if cant upload 1`] = `
<header
  class="row-content-block border-top-0 p-2 d-flex"
>
  <div>
    <!---->
     
    <input
      accept="image/*"
      class="hide"
      multiple="multiple"
      name="design_file"
      type="file"
    />
  </div>
</header>
`;

exports[`Design management upload form component renders loading icon 1`] = `
<header
  class="row-content-block border-top-0 p-2 d-flex"
>
  <div>
    <glbutton-stub
      class="btn-inverted"
      disabled="true"
      variant="primary"
    >
      
      Upload designs
      
      <glloadingicon-stub
        class="ml-1"
        color="orange"
        inline="true"
        label="Loading"
        size="sm"
      />
    </glbutton-stub>
     
    <input
      accept="image/*"
      class="hide"
      multiple="multiple"
      name="design_file"
      type="file"
    />
  </div>
</header>
`;

exports[`Design management upload form component renders upload design button 1`] = `
<header
  class="row-content-block border-top-0 p-2 d-flex"
>
  <div>
    <glbutton-stub
      class="btn-inverted"
      variant="primary"
    >
      
      Upload designs
      
      <!---->
    </glbutton-stub>
     
    <input
      accept="image/*"
      class="hide"
      multiple="multiple"
      name="design_file"
      type="file"
    />
  </div>
</header>
`;
